src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs

```diff
@@ -9,23 +9,12 @@ namespace Microsoft.CodeAnalysis.CSharp
 {
     internal static class BestTypeInferrer
     {
-        public static TypeSymbolWithAnnotations InferBestType(ImmutableArray<TypeSymbolWithAnnotations> types, Conversions conversions, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        public static TypeSymbol InferBestType(ArrayBuilder<TypeSymbol> types, ConversionsBase conversions, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
```

Comment:
>InferBestType [](start = 33, length = 13)

Might it be useful to add an assertion that the result doesn't depend on order of inputs? We could use reverse order, or maybe a randomized order for comparison.

Reply:
We're testing different order of inputs.

---
In reply to: [223857306](https://github.com/dotnet/roslyn/pull/29727#discussion_r223857306) [](ancestors = 223857306)